### DISCUSSION TITLE
How to hide DEBUG:parse:format messages

### DISCUSSION THREAD

**Discussion by MNITD** [behave/behave#946] [Q&A] [answered]
Hello!

I was trying to hide messages of format `DEBUG:parse:format '{something}' -> '(?P<something>.+?)'` during running behave

I used different options of `--logging-level` flag but seems like nothing helps 

Can anybody help with that issue?

**Comment by jenisys** [comment, marked as answer]
The output seems to originate from the `parse` module in the constructor of the `Parser` class (`logger=“parse“`).
I have never seen such an output.

Are you seeing this output because a scenario fails ?
Have you otherwise adapted the setup of the logging subsystem ?

NOTES:
* The logging module is rather picky when the basic setup is used (can only be called once, additional calls are ignored)
* Logging subsystem should be set up under control of `behave`
* `behave.ini` normally sets the default logging=INFO . Therefore, I would never see DEBUG outputs.
* Normally, stdout and logging output is captured during the execution of scenario. Therefore, I assume that you have failing scenarios
* Output of hooks is normally not captured
* You should verify your solution that logging level disables the problem by using `behave —no-logcapture …`. This will show log records even if the scenario passes.

SEE ALSO: `logcapture*.feature`, `logging*.feature`
* [features/logcapture.feature](https://github.com/behave/behave/blob/master/features/logcapture.feature)
* [features/logging.no_capture.feature](https://github.com/behave/behave/blob/master/features/logging.no_capture.feature)
* [features/logging.setup_level.feature](https://github.com/behave/behave/blob/master/features/logging.setup_level.feature)

**Comment by jenisys** [comment]
One additional hint: You will not see any logging output if it is redirected to a file (instead of console / stdout).

**Comment by MNITD** [comment]
@jenisys Thank you

You are right the messages were from parse module https://github.com/r1chardj0n3s/parse/blob/master/parse.py#L840

Also, I tried different config settings but it didn't affect logging in any way.

Later I found that the problem was in my project source code and exactly `logging.basicConfig(level=logging.DEBUG)` that overrrided global logging
